networking/metallb/metallb-configure-bgp-peers.adoc

```diff
@@ -0,0 +1,183 @@
+[id="metallb-configure-bgp-peers"]
+= Configuring MetalLB BGP peers
+include::modules/common-attributes.adoc[]
+:context: configure-metallb-bgp-peers
+
+toc::[]
+
+As a cluster administrator, you can add, modify, and delete Border Gateway Protocol (BGP) peers.
+The MetalLB Operator uses the BGP peer custom resources to identify which peers that MetalLB contacts to start BGP sessions.
+The peers receive the route advertisements for the load-balancer IP addresses that MetalLB assigns to services.
+
+// Dear reviewers and maintainers, I capitalized Autonomous System (AS)
+// to match the capitalization that is shown in the common terms section
+// of the foll RFC: https://datatracker.ietf.org/doc/html/rfc4271
+
+// BGP peer custom resource
+== About the BGP peer custom resource
+
+The fields for the BGP peer custom resource are described in the following table.
+
+.MetalLB BGP peer custom resource
+[cols="1,1,3",options="header"]
+|===
+
+|Field
+|Type
+|Description
+
+|`metadata.name`
+|`string`
+|Specifies the name for the BGP peer custom resource.
+
+|`metadata.namespace`
+|`string`
+|Specifies the namespace for the BGP peer custom resource.
+
+|`spec.myASN`
+|`integer`
+|Specifies the Autonomous System number for the local end of the BGP session.
+The range is `0` to `65535`.
+
+|`spec.peerASN`
+|`integer`
+|Specifies the Autonomous System number for the remote end of the BGP session.
+The range is `0` to `65535`.
+
+|`spec.peerAddress`
+|`string`
+|Specifies the IP address of the peer to contact for establishing the BGP session.
+
+|`spec.sourceAddress`
+|`string`
+|Optional: Specifies the IP address to use when establishing the BGP session.
+
+|`spec.peerPort`
+|`integer`
+|Optional: Specifies the network port of the peer to contact for establishing the BGP session.
+The range is `0` to `16384`.
+
+|`spec.holdTime`
+|`integer`
+|Optional: Specifies the duration, in seconds, for the hold time to propose to the BGP peer.
+The minimum value is `3` seconds.
+To detect path failures more quickly, also configure BFD.
+
+|`spec.keepaliveTime`
+|`integer`
+|Optional: Specifies the maximum interval, in seconds, between sending keep-alive messages to the BGP peer.
+If you specify this field, you must also specify a value for the `holdTime` field.
+The specified value must be less than the value for the `holdTime` field.
```

Comment:
The type of parameters `spec.holdTime` and `spec.keepaliveTime` changed to `string` (converts to `metav1.Duration` in the backend). Examples: `3s`, `1m`, `5m30s`.
Upstream PR: https://github.com/metallb/metallb-operator/pull/132